kairos-cli/bin/crypto/signer.rs

```diff
@@ -0,0 +1,54 @@
+use std::path::PathBuf;
+
+use super::private_key::{parse_private_key, CasperPrivateKey};
+use super::public_key::CasperPublicKey;
+use crate::crypto::error::CryptoError;
+use casper_types::{bytesrepr::ToBytes, SecretKey};
+use casper_types::{crypto, PublicKey};
+
+pub struct CasperSigner {
+    secret_key: CasperPrivateKey,
+    public_key: CasperPublicKey,
+}
+
+#[allow(unused)]
+impl CasperSigner {
+    fn from_key_raw(secret_key: CasperPrivateKey) -> Self {
+        // Derive the public key.
+        let public_key = CasperPublicKey::from_key(PublicKey::from(&secret_key.0));
+
+        CasperSigner {
+            secret_key,
+            public_key,
+        }
+    }
+
+    pub fn from_file(secret_key_path: &str) -> Result<Self, CryptoError> {
+        let secret_key =
+            SecretKey::from_file(secret_key_path).map_err(|_e| CryptoError::FailedToParseKey)?;
+
+        Ok(Self::from_key_raw(CasperPrivateKey(secret_key)))
+    }
+
+    pub fn from_key_pathbuf(secret_key_path: PathBuf) -> Result<Self, CryptoError> {
+        let private_key_path_str: &str = secret_key_path.to_str().ok_or(CryptoError::KeyLoad)?;
+        let private_key = parse_private_key(private_key_path_str)?;
+
+        Ok(Self::from_key_raw(private_key))
+    }
+
+    pub fn get_public_key(&self) -> CasperPublicKey {
+        self.public_key.clone()
+    }
```

Comment:
This hides a clone, just let the user access it throught the publicly exposed attribute and let the user decide what to do.

Reply:
Done in https://github.com/cspr-rad/kairos/pull/17/commits/8b78660571180b15f5319049925eb6431b0ced1d.